internal/core/textlayout/sharedparley.rs

```diff
@@ -0,0 +1,142 @@
+// Copyright © SixtyFPS GmbH <info@slint.dev>
+// SPDX-License-Identifier: GPL-3.0-only OR LicenseRef-Slint-Royalty-free-2.0 OR LicenseRef-Slint-Software-3.0
+
+pub use parley;
+
+use std::cell::RefCell;
+
+use crate::{
+    graphics::FontRequest,
+    items::TextStrokeStyle,
+    lengths::{LogicalLength, PhysicalPx},
+    textlayout::{TextHorizontalAlignment, TextOverflow, TextVerticalAlignment, TextWrap},
+};
+use i_slint_common::sharedfontique;
+
+pub const DEFAULT_FONT_SIZE: LogicalLength = LogicalLength::new(12.);
+
+fn font_context() -> parley::FontContext {
+    parley::FontContext {
+        collection: sharedfontique::COLLECTION.inner.clone(),
+        source_cache: sharedfontique::COLLECTION.source_cache.clone(),
+    }
+}
+
+std::thread_local! {
+    static LAYOUT_CONTEXT: RefCell<parley::LayoutContext<Brush>> = Default::default();
+}
```

Comment:
Consider boxing the `LayoutContext` so it doesn't take up as much space for unrelated threads.